src/reanimated2/hook/useScrollViewOffset.ts

```diff
@@ -15,23 +21,49 @@ const scrollEventNames = [
   'onMomentumScrollEnd',
 ];
 
+const addListenerToScroll = (
+  offsetRef: ScrollSharedValue<number>,
+  animatedRef: any
+) => {
+  runOnUI(() => {
+    'worklet';
+    offsetRef.triggerScrollListener = true;
+    offsetRef.addListener(animatedRef(), (newValue: any) => {
```

Comment:
Would be good to make sure we setup the listener only once. This code will be executed on every render for components using `useScrollViewOffset` hook. There are already some useEffect calls in that hook, so maybe we should put this initialization code there?